Add switch to OrgUnitsSelector to use orgUnit.shortName

jordibc opened the pull request:
Reference ticket: https://app.clickup.com/t/2ghev8r

We want the OU tree to be able to show shortnames instead of just the
default displaynames. This commit adds a Switch inside a FormControlLabel
between the search box and the OU tree of the OrgUnitsSelector
component to do just that.

The trickiest part was adding:
  key={useShortNames}
to the OrgUnitTree component, because it not having a unique key before
meant that react never knew it had to update it after changing the
state of the OrgUnitsSelector component.

jordibc committed:
Add swith to OrgUnitsSelector to use shortnames.

Reference ticket: https://app.clickup.com/t/2ghev8r

We want the OU tree to be able to show shortnames instead of just the
default displaynames. This commit adds a Switch inside a FormControlLabel
between the search box and the OU tree of the OrgUnitsSelector
component to do just that.

The trickiest part was adding:
  key={useShortNames}
to the OrgUnitTree component, because it not having a unique key before
meant that react never knew it had to update it after changing the
state of the OrgUnitsSelector component.

jordibc commented:
This is how it looks now, deactivated:

![Screenshot from 2022-07-08 10-03-21](https://user-images.githubusercontent.com/1568405/177949164-b23af2b2-be8c-4553-b068-20d01e729990.png)

And activated (note that the shortname is the same for these org units, except the root one, which doesn't have one):

![Screenshot from 2022-07-08 10-03-35](https://user-images.githubusercontent.com/1568405/177949191-560be454-f4a7-45e5-8e7a-6a07d798b811.png)

tokland commented on src/org-units-selector/OrgUnitsSelector.jsx:
```diff
@@ -1,4 +1,4 @@
-import { Card, CardContent } from "@material-ui/core";
```
This file is not prettified (`yarn prettify`). Having vscode with extension `Prettier` + setting "Format on save" enabled should solve the problem altogether.

jordibc replied:
Right, and set it as the default formatter in the general preferences. I changed my programming setup and missed it, even though you had already showed me this. Fixed now. Thanks!

tokland commented on i18n/en.pot:
```diff
@@ -180,3 +180,6 @@ msgstr ""
 msgstr ""
+
+msgid "Use short names"
+msgstr ""
```
Running `yarn localize` moves these new translation in the file, does it happen also for you?

jordibc replied:
It does happen to me. I wasn't that familiar with the proper use of localize, so I missed it. I understand now that one should run it too when adding new texts. Fixed now, thanks!

jordibc replied:
Ref: https://github.com/EyeSeeTea/d2-ui-components/commit/83b3c36285

tokland commented on src/org-unit-tree/OrgUnitTree.component.js:
```diff
@@ -50,6 +50,7 @@ class OrgUnitTree extends React.Component {
         super(props);
 
         this.state = {
+            useShortNames: props.useShortNames,
```
`useShortNames` is not updated within the component, right? then it can be simply used from the props.

jordibc replied:
Totally. Done in https://github.com/EyeSeeTea/d2-ui-components/commit/62085fc06ed

tokland commented on src/org-units-selector/OrgUnitsSelector.jsx:
```diff
@@ -312,7 +321,20 @@ export default class OrgUnitsSelector extends React.Component {
                         <div style={styles.contentItem}>
                             {roots.map(root => (
                                 <div key={root.path} className={`ou-root ${getClass(root)}`}>
+                                    <FormControlLabel
```
This new switch will now always be visible to users. However, in most cases, we will want to keep it as-is, a lot of projects use this component. Proposal: Use two props for `OrgUnitTree` so the caller has more control:

- `showShortName?: boolean` (defaults to false)
- `showNameSetting?: boolean` (defaults to false)

jordibc replied:
Makes total sense. Fixed in https://github.com/EyeSeeTea/d2-ui-components/commit/97676e17

jordibc committed:
Prettify.

Ref: https://github.com/EyeSeeTea/d2-ui-components/pull/227/files#r916687385

jordibc committed:
useShortNames directly from the props.

The property useShortNames is not updated within the component,
so it can be simply used from the props.

Ref: https://github.com/EyeSeeTea/d2-ui-components/pull/227/files#r916690401

jordibc committed:
Translations sorted with yarn localize.

Ref: https://github.com/EyeSeeTea/d2-ui-components/pull/227#discussion_r916688532

jordibc committed:
Add showShortName propery to OrgUnitSelector.

This property controls whether we show the Switch to choose
between displayNames and shortNames for the org units.

By default, it is not shown, as per its most common usage
and compatibility with previous uses of the component.

Ref: https://github.com/EyeSeeTea/d2-ui-components/pull/227#discussion_r916692514

jordibc commented:
Thanks @tokland , I found all the points valid and I think they are all addressed with the last commits, so I'm sending a new request for review.

jordibc committed:
Both properties showShortName and showNameSetting for OrgUnitsSelector.

The two properties are:
* showShortName - if true, defaults to use shortNames instead of displayNames
* showNameSetting - if true, shows a Switch to control if we want shortNames or displayNames

This fixes my previous commit, where I misunderstood the comment at
ref https://github.com/EyeSeeTea/d2-ui-components/pull/227#discussion_r916692514

jordibc commented:
Just to let you know, I tried to use the component from user-extended-app, with:

```html
<OrgUnitsSelector
    ...
    showShortName={xxx}
    showNameSetting={yyy}
/>
```

for all true/false combinations of `xxx` and `yyy` and it seems to work.

tokland reviewed (approved):
Cool, all good by me!

jordibc committed:
Bump version to 2.6.12-beta.3

ifoche reviewed (approved):
thanks @jordibc  @tokland
